### DISCUSSION TITLE
Replace url parameter without changing the others

### DISCUSSION THREAD

**Discussion by ToptenDev** [community/community#54042] [Programming Help] [answered]
### Select Topic Area

Question

### Select Feature Area

- [X] Issues
- [ ] Projects

### Body

I want to dynamically change a specific parameter in the navigation bar. I have this url and want to change "pedro" by "max":

http://mywebsite.com/?par1=mynameis&par2=pedro&par3=nicetomeetyou

By using this, it removes all my parameters.

hello = "max"
window.history.replaceState(null, null, "?par2="+hello);

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by Krieit** [comment, marked as answer]
Use the [URL interface](https://developer.mozilla.org/en-US/docs/Web/API/URL)

```
const hello = 'max';
// let url = new URL(location.href); // from location.href
let url = new URL(`http://mywebsite.com/?par1=mynameis&par2=pedro&par3=nicetomeetyou`); // from string
url.searchParams.set('par2',hello); // change ONE parameter
console.log(url.searchParams.toString());
window.history.replaceState(null, null, `?${url.searchParams.toString()}`);
```
